api/agent/drivers/driver.go

```diff
@@ -75,6 +75,9 @@ type Driver interface {
 	// Set image pull retry policy and retriable error checker
 	SetPullImageRetryPolicy(policy common.BackOffConfig, checker RetryErrorChecker) error
 
+	// Get fields from call extensions that are used in finding a slot to place the call
+	GetSlotKeyExtensions(extn map[string]string) map[string]string
```

Comment:
I think this can simply return a hash (aka string) to keep things simple. Then getSlotQueueKey() wouldn't need to handle sorting, etc.

Reply:
changed to return a hash (serialized string of the required fileds in extn)